sql/src/main/java/io/crate/execution/dml/ShardResponse.java

```diff
@@ -91,10 +95,43 @@ public String toString() {
 
     private IntArrayList locations = new IntArrayList();
     private List<Failure> failures = new ArrayList<>();
+
+    /**
+     * Result rows are used to return values from updated rows.
+     */
+    @Nullable
+    private List<Object[]> resultRows;
+
+    /**
+     * Result columns describe the types of the result rows.
+     */
+    @Nullable
+    private Symbol[] resultColumns;
+
+    /**
+     * Streamers to stream return values from updated rows.
+     */
+    @Nullable
+    private Streamer[] resultRowStreamers;
```

Comment:
Would it make sense to keep these only in local scope and derive them from the resultColumns as needed? Would make the relation-ship between resultColumns and resultRowStreamers clearer (that one is derived from the other)

There is also a `Symbols.streamerArray` function which can be used to do so.